packages/block-editor/src/components/use-block-commands/index.js

```diff
@@ -283,6 +292,37 @@ const getQuickActionsCommands = () =>
 			} );
 		}
 
+		if ( canToggleBlockVisibility ) {
+			const hasHiddenBlock = blocks.some(
+				( block ) =>
+					block.attributes.metadata?.blockVisibility === false
+			);
+
+			commands.push( {
+				name: 'core/toggle-block-visibility',
+				label: hasHiddenBlock ? __( 'Show' ) : __( 'Hide' ),
+				callback: () => {
+					const attributesByClientId = Object.fromEntries(
+						blocks?.map( ( { clientId, attributes } ) => [
+							clientId,
+							{
+								metadata: cleanEmptyObject( {
+									...attributes?.metadata,
+									blockVisibility: hasHiddenBlock
+										? undefined
+										: false,
+								} ),
+							},
+						] )
+					);
+					updateBlockAttributes( clientIds, attributesByClientId, {
+						uniqueByBlock: true,
+					} );
+				},
+				icon: hasHiddenBlock ? unseen : seen,
```

Comment:
```suggestion
				icon: hasHiddenBlock ? seen : unseen,
```
Nit: I think the icon should indicate the state of the block as a result of an action, not its current state, which means they should match the meaning of the label text.

| Before | After |
|--------|--------|
| <img width="418" height="199" alt="Image" src="https://github.com/user-attachments/assets/d0572095-3185-4524-948b-233247f07690" /> | <img width="424" height="206" alt="Image" src="https://github.com/user-attachments/assets/1c6df358-9568-4865-9a8c-7c34d4903968" /> |

Reply:
thanks for spotting!